language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -475,6 +480,30 @@ void setClientCapabilities(LSClientCapabilities clientCapabilities) {
         });
     }
 
+    @Override
+    public CompletableFuture<Either<Range, PrepareRenameResult>> prepareRename(PrepareRenameParams params) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                ReferencesContext context = ContextBuilder.buildReferencesContext(params.getTextDocument().getUri(),
```

Comment:
Shall we create a separate context type as PrepareRenameContext? It's true it will be the same but that will avoid future API hierarchy changes. Same applies for Rename operation as well.

Reply:
Done!